### ISSUE TITLE
node.parent?._dirty != true is not true hit in iOS simulator

### ISSUE THREAD

**Issue by eseidelGoogle** [flutter/flutter#7861]
I have no idea what I did to cause this.  I just was running the gallery in the iOS simulator.

```
══╡ EXCEPTION CAUGHT BY SCHEDULER LIBRARY ╞═════════════════════════════════════════════════════════
The following assertion was thrown during a scheduler callback:
'package:flutter/src/rendering/semantics.dart': Failed assertion: line 675 pos 14:
'node.parent?._dirty != true': is not true.
Either the assertion indicates an error in the framework itself, or we should provide substantially
more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new
When the exception was thrown, this was the stack:
#2      SemanticsOwner.sendSemanticsUpdate (package:flutter/src/rendering/semantics.dart:675:14)
#3      PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:1089:23)
#4      BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding.beginFrame (package:flutter/src/rendering/binding.dart:228:19)
#5      BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding&WidgetsBinding.beginFrame (package:flutter/src/widgets/binding.dart:327:23)
#6      BindingBase&SchedulerBinding&GestureBinding&ServicesBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:169:5)
#7      BindingBase&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:676:15)
#8      BindingBase&SchedulerBinding.handleBeginFrame (package:flutter/src/scheduler/binding.dart:611:9)
#9      _beginFrame (file:///b/build/slave/Mac_Engine/build/src/flutter/lib/ui/hooks.dart:83)
(elided 2 frames from class _AssertionError)
════════════════════════════════════════════════════════════════════════════════════════════════════
```

**Comment by eseidelGoogle** [comment]
flutter/flutter@cce70d706988ecda7cf93f294d157651b016b078

**Comment by abarth** [comment]
The iOS simulator runs with AX turned on, which is why you're seeing this AX code get run.

**Comment by abarth** [comment]
I've seen this before, but I'm unable to reproduce it now.  Closing for lack of a repro case.  Please re-open if you know how to reproduce the issue.

**Comment by kdy1** [comment]
@abarth This issue occurs when invoking calllback from a widget causes re-building the widget.

This can be fixed by caching instance of Widget in class field.


Example


```dart
class Callback extends StatefulWidget {
    final VoidCallback callback;
    @override
    CallbackState  createState() => CallbackState ();
}
class CallbackState extends State<Callback> {
  // ..

  // call this somehow (with something like onPressed)
  void onEvent(){
     widget.callback()
  }
}


class Usage extends StatefulWidget {
    @override
    UsageState createState() => UsageState ();
}
class UsageState extends State<Usage> {
    @override
    Widget build(BuildContext context) {
        return Callback(
            callback: () {
                setState(() { });
            },
        );
    }

}

```
